### PR TITLE
@vocdoni/rainbowkit-wallets

### PR DESCRIPTION
**Pull request by nigeon**
Rainbowkit wallets & wagmi connectors for easily extending web3 connectivity.
- InputsWallet: Asks for 2 input fields and generates a deterministic wallet based on them
- OAuth Wallet it uses an external service (https://github.com/vocdoni/oauth-service) in order to authentify a user and generates a wallet based on that.

### CODE REVIEW

File: packages/rainbowkit-wallets/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "dev": "NODE_ENV=development yarn build --watch",
+    "lint": "tsc",
+    "build": "vite build",
+    "size": "size-limit",
+    "preview": "vite preview",
+    "prepack": "json -f package.json -I -e \"delete this.devDependencies; delete this.dependencies\""
```
Comment (elboletaire) [inline packages/rainbowkit-wallets/package.json:22]: I'm not sure you should be doing this. Instead, I'd add the dependencies that have to be bundled to the "devDependencies", and would remove only that (leaving the dependencies intact, with dependencies that will not be bundled [aka installed externally]).

File: packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    setIsOpen(false)
+    props.onSubmit({ username, password })
+  }
+
+  return (
+    <div>
```
Comment (elboletaire) [inline packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx:39]: There should be no need for this parent `<div>`. Also, if you want just to wrap components to export a single component, use fragments instead (`<></>`).

File: packages/rainbowkit-wallets/src/lib/localStorageWallet.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  static storageItemName = 'localstorage-wallet-seed'
+
+  public static async getWallet(): Promise<Wallet | undefined> {
+    try {
+      const value: string = localStorage.getItem(this.storageItemName) as string
+      if (!value) return undefined
+
+      return this.createWallet(value)
+    } catch (err) {
+      console.log(err)
```
Comment (elboletaire) [inline packages/rainbowkit-wallets/src/lib/localStorageWallet.ts:14]: Better use console.error or console.warn for these cases. It's also a good idea to prefix it with a specific string, so we can easily locate it later (i.e. `console.error('failed to generate wallet:', err)`).

**Review by elboletaire — CHANGES_REQUESTED** [review]
Diagonal review (haven't checked everything since you said it's still a draft).

File: packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx (hunk 2 above)
Comment (nigeon) [reply packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx:39]: Sure. isn't <></> the short syntax for React.Fragment (https://legacy.reactjs.org/docs/fragments.html#short-syntax)?

Comment (elboletaire) [reply packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx]: That's why I said "use fragments instead" 😅

Comment (nigeon) [reply packages/rainbowkit-wallets/src/components/CreateWalletModal.tsx]: not sure what I read xD

**Review by elboletaire — APPROVED** [review]
I still need to properly check its integration (I'm currently on it), but looking good so far.

**Pull request merged**